apps/framework-cli/src/cli.rs

```diff
@@ -76,25 +78,29 @@ async fn top_command_handler(settings: Settings, commands: &Commands) {
                     exit(1);
                 }
 
-                let project = Project::from_dir(dir_path, name.clone(), *language);
+                let project = Project::new(dir_path, name.clone(), *language);
+                let project_arc = Arc::new(project);
```

Comment:
I guess the change comes from that we are having too many `project.clone()`. I wonder if we can avoid cloning and reference counting by adding lifetime to the structs.

Not saying that I want this done in this PR.

Reply:
Hum that's a good question. Yeah this is allowing the same instance to be passed around across threads without cloning.